### PR TITLE
implement division and some helper functions

### PR DESCRIPTION
**Pull request by remdragon**

### CODE REVIEW

File: BigInt.c [hunk 1]
```diff
@@ -119,6 +119,27 @@ BigInt* BigInt_construct(int value) {
     return new_big_int;
 }
 
+BigInt* BigInt_clone(const BigInt* big_int, unsigned int num_allocated_digits) {
+    if(num_allocated_digits > big_int->num_digits) {
```
Comment (dandclark) [inline BigInt.c]: I don't understand the reason for the `num_allocated_digits` parameter. If it's bigger than `big_int->num_digits` then we snap it to `big_int->num_digits`, but if it's smaller than `big_int->num_digits`, we'll end up creating `new_big_int` with `num_allocated_digits` < `num_digits` which is incorrect.

The callers are all passing `divisor->num_digits + 1` which will just get snapped to `divisor->num_digits` anyway, so can we just remove the parameter?

**Review by dandclark — COMMENTED** [review]
Looks good, just one question about the extra parameter in `BigInt_clone`.

File: BigInt.c (hunk 1 above)
Comment (remdragon) [reply BigInt.c]: Just fixed it, the comparison was accidentally backwards. The purpose of the parameter is to prevent a likely reallocation.

**Comment by dandclark** [comment]
Thanks for the PR!

**Pull request merged**